app/renderer/components/bookmarksToolbar.js

```diff
@@ -47,6 +48,21 @@ class BookmarkToolbarButton extends ImmutableComponent {
     }
   }
 
+  onMouseOver (e) {
+    // Behavior when a bookmarks toolbar folder has its list expanded
+    if (this.props.selectedFolderId) {
```

Comment:
I was looking at `this.props.showBookmarkFolderMenu` which at first looked like it called the electron api directly, but it calls `setContextMenuDetails` which eventually triggers a call the remote api. The windowStore really should directly care about any of this, the logic itself should be in a "reducer" similar `extensionState`. In the flux/redux model that is exactly where the logic is supposed to go. The UI should just fire off events and react to state changes. We have a bit of code to untangle because actions shouldn't call other actions, but some of the actions are "setters" and aren't written in a way that makes it easy to reuse the login in them.